src/test/java/hudson/plugins/validating_parameters/password/ValidatingPasswordParameterDefinitionTest.java

```diff
@@ -0,0 +1,272 @@
+        // given
+        String validInput = "val";
+        mockSecretFor(validInput);
+
+        // when
+        ValidatingPasswordParameterDefinition d =
+                new ValidatingPasswordParameterDefinition(
+                        DEF_NAME, DEF_DEFAULT_VALUE, DEF_REGEX, DEF_MESSAGE, DEF_DESCRIPTION);
+        ParameterValue value = d.createValue(validInput);
+
+        // then
+        assertThat(value).isNotNull();
+        assertThat(value).isInstanceOf(ValidatingPasswordParameterValue.class);
+
+        ValidatingPasswordParameterValue specificValue = (ValidatingPasswordParameterValue)value;
+        assertThat(specificValue.getValue().getPlainText()).isEqualTo(validInput);
+        assertThat(specificValue.getName()).isEqualTo(DEF_NAME);
+        assertThat(specificValue.getDescription()).isEqualTo(DEF_DESCRIPTION);
+        assertThat(specificValue.getRegex()).isEqualTo(DEF_REGEX);
+    }
+
+    @Test
+    public void shouldReturnDefaultValueFromNull() {
+        // given / when
+        ValidatingPasswordParameterDefinition d =
+                new ValidatingPasswordParameterDefinition(
+                        DEF_NAME, DEF_DEFAULT_VALUE, DEF_REGEX, DEF_MESSAGE, DEF_DESCRIPTION);
+        ParameterValue value = d.createValue((String)null);
+
+        // then
+        assertThat(value).isNotNull();
+        assertThat(value).isInstanceOf(ValidatingPasswordParameterValue.class);
+
+        ValidatingPasswordParameterValue specificValue = (ValidatingPasswordParameterValue)value;
+        assertThat(specificValue.getValue().getPlainText()).isEqualTo(DEF_DEFAULT_VALUE);
+    }
+
+    @Test
+    public void shouldReturnDefaultValueFromBlankString() {
+        // given / when
+        ValidatingPasswordParameterDefinition d =
+                new ValidatingPasswordParameterDefinition(
+                        DEF_NAME, DEF_DEFAULT_VALUE, DEF_REGEX, DEF_MESSAGE, DEF_DESCRIPTION);
+        ParameterValue value = d.createValue("");
+
+        // then
+        assertThat(value).isNotNull();
+        assertThat(value).isInstanceOf(ValidatingPasswordParameterValue.class);
+
+        ValidatingPasswordParameterValue specificValue = (ValidatingPasswordParameterValue) value;
+        assertThat(specificValue.getValue().getPlainText()).isEqualTo(DEF_DEFAULT_VALUE);
+    }
+
+    @Test
+    public void shouldThrowExceptionForRegexMismatchingString() {
+        // given
+        String invalidInput = "000";
+        mockSecretFor(invalidInput);
+
+        // when
+        ValidatingPasswordParameterDefinition d =
+                new ValidatingPasswordParameterDefinition(
+                        DEF_NAME, DEF_DEFAULT_VALUE, DEF_REGEX, DEF_MESSAGE, DEF_DESCRIPTION);
+
+        // then
+        assertThatThrownBy(() -> d.createValue(invalidInput))
+                .isInstanceOf(Failure.class)
+                .hasMessage(DEF_MESSAGE);
+    }
+
+    @Test
+    public void shouldCreateValueFromStaplerRequestAndJSONWithRegexMatchingString() {
```

Comment:
It seems that some of test could be parametrised

Reply:
Let's do this next time. Now we would not gain to many benefits.